Add greek translations for Measurement provider & other Greek tests

gvrettos opened the pull request:
Some more translations, for Measurement provider this time.

Could you please annotate this PR with **hacktoberfest-accepted** label?

kingthorin reviewed (commented):
Could you add tests?

gvrettos commented:
> Could you add tests?
Hi @kingthorin!
Do you have something specific in mind? `MeasurementTest` and `FakerIntegrationTest` seem relevant but I could not find something that needs updating. Do I miss something?


> Hi @gvrettos , thank you for your contribution. Feel free to send them, but we won't be applying `hacktoberfest` labels to your PRs anymore. Thank you.
Hi @bodiam! I am sorry if that got too frustrating for you. ☹️  
Could you please let me know what is the reason for not applying the `hacktoberfest` label? I would like to know if I have done something wrong here.

kingthorin commented:
How about something like this?
```diff
diff --git a/src/test/java/net/datafaker/providers/base/BaseFakerTest.java b/src/test/java/net/datafaker/providers/base/BaseFakerTest.java
index 4f4a16d8..c3d438ba 100644
--- a/src/test/java/net/datafaker/providers/base/BaseFakerTest.java
+++ b/src/test/java/net/datafaker/providers/base/BaseFakerTest.java
@@ -22,7 +22,7 @@
 public class BaseFakerTest<T extends BaseFaker> {
 
     private static final Logger LOG = Logger.getLogger(BaseFakerTest.class.getCanonicalName());
-    protected final T faker = getFaker();
+    protected T faker = getFaker();

     @BeforeEach
     @SuppressWarnings("EmptyTryBlock")
@@ -38,6 +38,11 @@ protected T getFaker() {
         return (T) new BaseFaker();
     }

+    @SuppressWarnings("unchecked")
+    protected void setFaker(T faker) {
+        this.faker = (T) faker;
+    }
+
     protected List<String> getBaseList(String key) {
         return faker.fakeValuesService().fetchObject(key, faker.getContext());
     }
diff --git a/src/test/java/net/datafaker/providers/base/MeasurementTest.java b/src/test/java/net/datafaker/providers/base/MeasurementTest.java
index 6c759c66..e88239d0 100644
--- a/src/test/java/net/datafaker/providers/base/MeasurementTest.java
+++ b/src/test/java/net/datafaker/providers/base/MeasurementTest.java
@@ -1,6 +1,12 @@
 package net.datafaker.providers.base;

 import java.util.List;
+import java.util.Locale;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+
 import java.util.Collection;

 class MeasurementTest extends BaseFakerTest<BaseFaker> {
@@ -19,4 +25,32 @@ protected Collection<TestSpec> providerListTest() {
             TestSpec.of(measurement::metricVolume, "measurement.metric_volume"),
             TestSpec.of(measurement::metricWeight, "measurement.metric_weight"));
     }
+
+    @Nested
+    class GreekMeasurementTest extends BaseFakerTest<BaseFaker> {
+
+        @BeforeAll
+        void setup() {
+            this.setFaker(new BaseFaker(new Locale("el", "GR")));
+        }
+
+        @AfterAll
+        void reset() {
+            this.setFaker(this.getFaker());
+        }
+
+        @Override
+        protected Collection<TestSpec> providerListTest() {
+
+            Measurement measurement = faker.measurement();
+            return List.of(TestSpec.of(measurement::height, "measurement.height"),
+                TestSpec.of(measurement::length, "measurement.length"),
+                TestSpec.of(measurement::volume, "measurement.volume"),
+                TestSpec.of(measurement::weight, "measurement.weight"),
+                TestSpec.of(measurement::metricLength, "measurement.metric_length"),
+                TestSpec.of(measurement::metricHeight, "measurement.metric_height"),
+                TestSpec.of(measurement::metricVolume, "measurement.metric_volume"),
+                TestSpec.of(measurement::metricWeight, "measurement.metric_weight"));
+        }
+    }
 }
```

Maybe also go back and add tests for your other addition too "wink:

gvrettos commented:
@kingthorin thank you very much for the boost with the tests. :+1:  
I also went back to any providers I had added support for Greek in the past and I added the respective tests there, too.

kingthorin commented on src/test/java/net/datafaker/providers/base/AncientTest.java:
```diff
@@ -1,16 +1,47 @@
+    }
+
+    private Collection<TestSpec> getProviderListTests(Ancient ancient) {
+        return List.of(
+            TestSpec.of(ancient::god, "ancient.god"),
             TestSpec.of(ancient::primordial, "ancient.primordial"),
             TestSpec.of(ancient::titan, "ancient.titan"),
-            TestSpec.of(ancient::hero, "ancient.hero"));
+            TestSpec.of(ancient::hero, "ancient.hero")
+        );
     }
```
Maybe move these to the top of the test classes?

gvrettos replied:
Done

kingthorin commented on src/test/java/net/datafaker/providers/base/AncientTest.java:
```diff
@@ -1,16 +1,47 @@
+        }
+    }
+
+    private Collection<TestSpec> getProviderListTests(Ancient ancient) {
```
Should be static

gvrettos replied:
Done
